src/App.tsx

```diff
@@ -1,25 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const items = getNumbers(1, 42).map(n => `Item ${n}`);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [perPage, setPerPage] = useState<number>(5);
+  const first = currentPage * perPage - perPage + 1;
+  const last = Math.min(items.length, currentPage * perPage);
+  const visibleItems = getNumbers(first, last);
```

Comment:
The `getNumbers` function is used here to calculate `visibleItems`, but it seems to be incorrect. It should generate the items based on the `items` array, not just numbers. Consider slicing the `items` array from `first - 1` to `last` to get the correct visible items for the current page.